kasa/interfaces/thermostat.py

```diff
@@ -0,0 +1,116 @@
+"""Interact with a TPLink Thermostat."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Annotated, Literal
+
+from ..feature import Feature
+from ..module import FeatureAttribute, Module
+
+
+class ThermostatState(Enum):
+    """Thermostat state."""
+
+    Heating = "heating"
+    Calibrating = "progress_calibration"
+    Idle = "idle"
+    Off = "off"
+    Unknown = "unknown"
+
+
+class Thermostat(Module, ABC):
+    """Base class for TP-Link Thermostat."""
+
+    def _initialize_features(self) -> None:
+        """Initialize features after the initial update."""
+        self._add_feature(
```

Comment:
Shouldn't these rather got to the thermostat module (like done for fan?), or just stay where they were?

Reply:
For `Fan` all the features are added from the `interface`. However for this case a `temperaturesensor` has a subset of the features so those ones need to be added from the module. it might be better to leave them all on the underlying modules as that's consistent with `light`.

Reply:
Now moved back to `termperaturecontrol`